test/functional/page_objects/visualize_page.js

```diff
@@ -206,7 +206,8 @@ export function VisualizePageProvider({ getService, getPageObjects }) {
     }
 
     toggleSpyPanel() {
-      return testSubjects.click('spyToggleButton');
+      testSubjects.click('spyToggleButton');
```

Comment:
I think these two statements probably race a little.

the click returns a promise so should be chained as such. Because we are sleeping for a second, the click probably just resolves in that waiting period.

so I don't think this would impact your change here, but I'd chain the sleep after the spy-toggle resolves.